Gems/AudioSystem/Code/Source/Engine/ATLAudioObject.cpp

```diff
@@ -594,18 +594,22 @@ namespace Audio
         // warn if there is activity on an object being cleared
         if (!m_cActiveEvents.empty())
         {
+            AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
             const char* const sEventString = GetEventIDs("; ").c_str();
             AZLOG_NOTICE(
                 "Events are active on an object (ID: %llu) being released!  #Events: %zu   EventIDs: %s", GetID(), m_cActiveEvents.size(),
                 sEventString);
+            AZ_POP_DISABLE_WARNING_GCC;
         }
 
         if (!m_cTriggers.empty())
         {
+            AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
```

Comment:
The call to GetTriggerNames, should remove the `.c_str()` call and assign the result to an AZStd::string.
```c++
const AZStd::string sTriggerString = GetTriggerNames("; ", pDebugNameStore);
```